lighthouse-core/gather/driver.js

```diff
@@ -1237,6 +1237,19 @@ class Driver {
     return flattenedDocument.nodes ? flattenedDocument.nodes : [];
   }
 
+  /**
+   * @param {number} nodeId
+   * @param {string} attributeName
+   * @param {string} attributeValue 
+   */
+  async setNodeAttribute(nodeId, attributeName, attributeValue) {
```

Comment:
IMO its fine to move this method back into your trace-nodes gatherer. and just inline it.